src/main/java/org/dcsa/ovs/service/impl/TimestampServiceImpl.java

```diff
@@ -43,6 +43,22 @@
 
     @Override
     public Mono<Timestamp> create(Timestamp timestamp) {
+        if (timestamp.getFacilitySMDGCode() == null) {
+            // OVS 2.0.0 Spec says optional, but our code does not function without it.  Let's be honest about it.
+            return Mono.error(new ResponseStatusException(HttpStatus.NOT_IMPLEMENTED));
+        }
+        if (timestamp.getPortCallServiceTypeCode() == null) {
+            // OVS 2.0.0 Spec says optional, operations event says mandatory.  The latter wins for now.
+            return Mono.error(new ResponseStatusException(HttpStatus.NOT_IMPLEMENTED));
+        }
+        if (timestamp.getModeOfTransport() == null) {
+            if (timestamp.getVesselIMONumber() == null) {
+                // OVS 2.0.0 Spec says optional, operations event says mandatory.  The latter wins for now.
+                return Mono.error(new ResponseStatusException(HttpStatus.NOT_IMPLEMENTED));
+            }
+            // Assume vessel IMO number implies VESSEL as mode of transport.
+            timestamp.setModeOfTransport(DCSATransportType.VESSEL);
+        }
```

Comment:
Now would be the time if we should update the API spec... I suppose this is only concerning OperationsEvents?

Reply:
I do not disagree but lets add it to "the list".